allennlp/commands/elmo.py

```diff
@@ -0,0 +1,203 @@
+                           for layer in layer_activations]
+        # Converts a list of pairs (activation, mask) tensors to a single tensor of activations.
+        activations = torch.cat([ele[0].unsqueeze(1) for ele in without_bos_eos], dim=1)
+        # The mask is the same for each ELMo vector, so just take the first.
+        mask = without_bos_eos[0][1]
+
+        return activations, mask
+
+    def embed_sentence(self, sentence: List[str]) -> torch.Tensor:
+        """
+        Computes the ELMo embeddings for a single tokenized sentence.
+        :param sentence: a tokenized sentence
+        :return: a tensor containing the ELMo vectors
+        """
+        return self.embed_batch([sentence])[0]
+
+    def embed_batch(self, batch: List[List[str]]) -> List[torch.Tensor]:
+        """
+        Computes the ELMo embeddings for a batch of tokenized sentences.
+        :param batch: a list of tokenized sentences
+        :return: a list of tensors, each representing the ELMo vectors for the input sentence at the same index
+        """
+        elmo_embeddings = []
+
+        embeddings, mask = self.batch_to_embeddings(batch)
+        for i in range(len(batch)):
+            length = int(mask[i, :].sum())
+            sentence_embeds = embeddings[i, :, :length, :].data.cpu().numpy()
+            elmo_embeddings.append(sentence_embeds)
+
+        return elmo_embeddings
+
+    def embed_sentences(self, sentences: Iterable[List[str]], batch_size: int) -> Iterable[torch.Tensor]:
+        """
+        Computes the ELMo embeddings for a iterable of sentences.
+        :param sentences: an iterable of tokenized sentences
+        :param batch_size: the number of sentences ELMo should process at once
+        :return: a list of tensors, each representing the ELMo vectors for the input sentence at the same index
+        """
+        for batch in lazy_groups_of(iter(sentences), batch_size):
+            yield from self.embed_batch(batch)
+
+    def embed_file(self,
+                   input_file: IO,
+                   output_file_path: str,
+                   batch_size: int,
+                   use_sentence_key: bool = False) -> None:
+        """
+        Computes ELMo embeddings from an input_file where each line contains a sentence tokenized by whitespace.
+        The ELMo embeddings are written out in HDF5 format, where each sentences is saved in a dataset corresponds
+        with a organized by key.  Unless use_sentence_key is set, the key will be the index of the sentence.
+        :param input_file: a file with one tokenized sentence per line
+        :param output_file_path: a path to the output hdf5 file
+        :param batch_size: the number of sentences to process in ELMo at one time
+        :param use_sentence_key: if true, use the first token in each sentence as the key
+        """
+
+        # Tokenizes the sentences.
+        sentences = [line.strip().split() for line in input_file]
+        if use_sentence_key:
```

Comment:
Hmm, when we discussed this I meant to use the entire sentence string as the key, but this is also fine. However, your wording here and in the docstring is confusing - it could be made clearer that the first token must be a unique identifier (rather than just the first token in the sentence), and then calling it `first_token_key` or something might be better.

Reply:
@DeNeutoy I have not implemented what we discussed.  This is the same behavior as in the original script.  I may change this in a future PR.

I will update the comments.